### ISSUE TITLE
[Proposal] dapr subscribe command

### ISSUE THREAD

**Issue by shubham1172** [dapr/cli#1127]
## Describe the proposal

This proposal is to add a `dapr subscribe` command, a counterpart for the existing `dapr publish` command. This will be useful in development scenarios which allows users to subscribe to a pubsub/topic/route, and print the data on STDOUT.

Example usage:

```bash
# Subscribe to sample topic in target pubsub via subscriber app without cloud event
dapr subscribe --subscribe-app-id myapp --pubsub target --topic sample --metadata '{"rawPayload":"true"}'
```

Proposed flags:

|Short|Long|Type|Usage|
|--|--|--|--|
| -m| --metadata| string |         The JSON serialized publish metadata (optional)|
| -i| --subscribe-app-id| string|    The ID of the subscriber app|
|-p| --pubsub |string       |   The name of the pub/sub component|
|-t| --topic| string  |  The topic to be published to|
|-u| --unix-domain-socket| string|   Path to a unix domain socket dir. If specified, Dapr API servers will use Unix Domain Sockets|
|-o| --output| string | Different levels of output - data (default), time, metadata, all |
|-r|--response| string | Response from the app - success (default), retry, drop |
|-i| --interactive| flag  | If specified, user is prompted to input the response for each message |

Future scope can also include a `--filter` option to selectively show data.

## Release Note

RELEASE NOTE: **ADD** Support for dapr subscribe

**Comment by mukundansundar** [comment]
@shubham1172 I think this is a proposal that will help users in quickly developing and testing pubsub API in Dapr.
With that I think we need to maintain consistency with other commands and have the `--subscribe-app-id` parameter shorthand as `-a`.

The one concern that I have with the `--routes` parameter is that it does not make much sense for a console output application. 
Unless say each event matching a particular match expression with a specific route is prefixed by that route in the output say
```
<datetime?> <rote.path> <event data> <metadata?>
```
?: optional
route.path : either default path or specific path 

WDYT?

**Comment by shubham1172** [comment]
>  Unless say each event matching a particular match expression with a specific route is prefixed by that route in the output say

@mukundansundar yes, that we can do. The main motivation for using a routes is the ability to filter events.

**Comment by philliphoff** [comment]
This feature would be useful in the development of Dapr pub-sub Pluggable Components, as the developer would not be forced to write an entire test harness or find and adapt an existing sample just to quickly test sending and, in particular, receiving events.
